### DISCUSSION TITLE
Assistant graph/plot retrieval

### DISCUSSION THREAD

**Discussion by dgedgafo** [RageAgainstThePixel/OpenAI-DotNet#300] [Q&A] [answered]
I'm having an issue where If an assistant returns a graph with a message I seem to get a fileID back instead of the python code I would expect. How can I use the fileID to retrieve the actual image of the graph? Here is a sample file id I receive: "file-8Z5BqwUHTy3IhRotxNVdts2k"

Thank you for your help!

**Comment by dgedgafo** [comment]
I should clarify that I get an ImageFile object back, but it seems like it only contains one property which is the FileID. How do I display an actual image from the ImageFile object?

**Reply by dgedgafo** [reply]
@StephenHodgson do you happen to have any idea what to do with the ImageFile object? What is the ID referring to? 

Thank you!

**Reply by StephenHodgson** [reply]
Usually the file id

**Reply by StephenHodgson** [reply, marked as answer]
You should be able to download the file using the file endpoint.

https://github.com/RageAgainstThePixel/OpenAI-DotNet?tab=readme-ov-file#download-file-content

**Reply by dgedgafo** [reply]
Thank you, that was it, I'm able to retrieve the file by ID. I notice that the assistant sometimes doesn't see the files that were uploaded to it. I can see the uploaded files in the assistants UI, however, if I try to run a query against the file the assistant will say that it doesn't have file with relevant data. Is that something you have encountered by chance?

**Reply by StephenHodgson** [reply]
A few times. Seems like an OpenAI bug
